linkerd/dns/src/lib.rs

```diff
@@ -26,15 +28,23 @@ pub trait ConfigureResolver {
 
 #[derive(Debug)]
 pub enum Error {
-    NoAddressesFound,
+    NoAddressesFound(Instant),
     ResolutionFailed(ResolveError),
     TaskLost,
 }
 
 pub type Task = Pin<Box<dyn Future<Output = ()> + Send + 'static>>;
 
+pub type ResolveResponseFuture =
+    Pin<Box<dyn Future<Output = Result<ResolveResponse, Error>> + Send + 'static>>;
```

Comment:
If the `Resolution` was implemented using `async-stream`, the `resolve_ips` function could just return `impl Future` (or even just be an `async fn`, I think). That would let us avoid the box here.